src/helpers/contracts.js

```diff
@@ -177,7 +177,7 @@ export const getFormInputsFromMethod = (
   formInputs,
   argOffset
 ) => {
-  if (!methodId) return;
+  if (!methodId || !abi) return formInputs;
```

Comment:
You're right, this can be simplified. I've moved this logic out to the useEffect where it is being called. Now `setFormInputs` will not be called unless we have updated a contract call method (ie. in `eth_call` and `trace_call`)

https://github.com/abunsen/etherflow/blob/5a09506e0e5ea5c2310afd67faebd7ec9d1b65bc/src/containers/methodCallContainer.js#L130